src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -0,0 +1,358 @@
+﻿using System;
+using Uno.Buffers;
+using System.Text;
+using Uno.Disposables;
+
+namespace Windows.Storage.Streams
+{
+	/// <summary>
+	/// Reads data from an input stream.
+	/// </summary>
+	public sealed partial class DataReader : IDataReader, IDisposable
+	{
+		private readonly static ArrayPool<byte> _pool = ArrayPool<byte>.Create();
+		private readonly IBuffer _buffer;
+
+		private int _bufferPosition = 0;
+
+		private DataReader(IBuffer buffer)
+		{
+			_buffer = buffer ?? throw new ArgumentNullException(nameof(buffer));
+		}
+
+		/// <summary>
+		/// Creates a new instance of the data reader with data from the specified buffer.
+		/// </summary>
+		/// <param name="buffer">The buffer.</param>
+		/// <returns>The data reader.</returns>
+		public static DataReader FromBuffer(IBuffer buffer) => new DataReader(buffer);
+
+		/// <summary>
+		/// Gets or sets the Unicode character encoding for the input stream.
+		/// </summary>
+		public UnicodeEncoding UnicodeEncoding { get; set; }
+
+		/// <summary>
+		/// Gets or sets the byte order of the data in the input stream.
+		/// </summary>
+		public ByteOrder ByteOrder { get; set; }
+
+		/// <summary>
+		/// Gets the size of the buffer that has not been read.
+		/// </summary>
+		public uint UnconsumedBufferLength => (uint)(_buffer.Length - _bufferPosition);
+
+		/// <summary>
+		/// Reads a byte value from the input stream.
+		/// </summary>
+		/// <returns>The value.</returns>
+		public byte ReadByte()
+		{
+			VerifyRead(1);
+			return ReadByteFromBuffer();
+		}
+
+		/// <summary>
+		/// Reads an array of byte values from the input stream.
+		/// </summary>
+		/// <param name="value">The array that receives the byte values.</param>
+		public void ReadBytes(byte[] value)
+		{
+			VerifyRead(value.Length);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'value' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)